apps/aehttp/src/aehttpc_doge.erl

```diff
@@ -57,59 +63,56 @@ hash_to_integer(Hash) ->
 get_chain_type() ->
     {ok, doge}.
 
-%%%=============================================================================
 %%% Pinning
+
+pin_to_pc({PinningData, _Who, Amount, Fee, _NetworkId, _SignModule}, NodeSpec) ->
+    PinPayload = aeser_hc:encode_parent_pin_payload(PinningData),
+    post_pin(NodeSpec, Amount, Fee, PinPayload).
+
+get_pin_by_tx_hash(TxHash, NodeSpec) ->
+    lager:debug("in doge with: ~p", [TxHash]),
+    {ok, #{<<"hex">> := RawTx, <<"confirmations">> := Conf} = Tx} = gettransaction(NodeSpec, TxHash),
+    lager:debug("raw_tx_hex: ~p", [RawTx]),
+    {ok, Vout} = decoderawtransaction(NodeSpec, RawTx),
+    {ok, Hex} = find_pin(Vout),
+    {ok, PinMap} = aeser_hc:decode_parent_pin_payload(aeu_hex:hex_to_bin(Hex)),
+    UpdMap =
+        case Conf of
+            0 -> maps:put(pc_height, -1, PinMap);
+            _ ->
+                Hash = maps:get(<<"blockhash">>, Tx),
+                {ok, {Height, _, _, _, _}} = getblock(NodeSpec, <<>>, Hash, 2),
+                maps:put(pc_height, Height, PinMap) % TODO fetch block using BlockHash from Tx
```

Comment:
Is this TODO needed? Don't you already fetch the block by hash?

Reply:
Yes, fixed that and forgot to take out. Will fix and stash for next PR :)